module/zfs/spa_log_spacemap.c

```diff
@@ -997,14 +997,21 @@ spa_ld_log_sm_metadata(spa_t *spa)
 
 	zap_cursor_t zc;
 	zap_attribute_t za;
-	for (zap_cursor_init(&zc, spa_meta_objset(spa), spacemap_zap);
-	    zap_cursor_retrieve(&zc, &za) == 0; zap_cursor_advance(&zc)) {
+	zap_cursor_init(&zc, spa_meta_objset(spa), spacemap_zap);
+	while ((error = zap_cursor_retrieve(&zc, &za)) == 0) {
```

Comment:
Why do this as a `while` loop vs a `for` loop?  I think the advantages of the `for` loop would still hold even when saving the `error`, and IIRC this form of zap iteration is used elsewhere:

```
for (zap_cursor_init(&zc, spa_meta_objset(spa), spacemap_zap);
    (error = zap_cursor_retrieve(&zc, &za)) == 0;
    zap_cursor_advance(&zc)) {
```

Reply:
Change applied.